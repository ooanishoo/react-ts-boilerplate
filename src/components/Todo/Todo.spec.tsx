import { cleanup, render } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { Todo } from '.'

afterEach(cleanup)
describe('<Todo/>', () => {
    test('should render without crashing', () => {
        const id = '3'
        const route = `/todos/${id}`
        window.history.pushState({}, 'Test page', route)
        const { container, getByTestId } = render(<Route path="/todos/:id" component={Todo} />, { wrapper: Router })
        expect(container).toMatchSnapshot()
        expect(getByTestId('todo')).toHaveTextContent(`This is a todo component with ID :${id}`)
    })
})
