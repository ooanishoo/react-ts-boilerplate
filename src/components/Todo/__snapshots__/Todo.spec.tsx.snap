// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo/> should render without crashing 1`] = `
<div>
  <div
    data-testid="todo"
  >
    This is a todo component with ID :
    3
  </div>
</div>
`;
